aptos-move/framework/aptos-framework/sources/zkid.move

```diff
@@ -0,0 +1,123 @@
+module aptos_framework::zkid {
+    use std::option;
+    use std::option::Option;
+    use std::signer;
+    use aptos_framework::system_addresses;
+
+    #[resource_group(scope = global)]
+    struct Group {}
+
+    #[resource_group_member(group = aptos_framework::zkid::Group)]
+    /// The 288-byte Groth16 verification key (VK) for the zkID relation.
+    struct Groth16VerificationKey has key, store {
+        /// 32-byte serialization of `alpha * G`, where `G` is the generator of `G1`.
+        alpha_g1: vector<u8>,
+        /// 64-byte serialization of `alpha * H`, where `H` is the generator of `G2`.
+        beta_g2: vector<u8>,
+        /// 64-byte serialization of `gamma * H`, where `H` is the generator of `G2`.
+        gamma_g2: vector<u8>,
+        /// 64-byte serialization of `delta * H`, where `H` is the generator of `G2`.
+        delta_g2: vector<u8>,
+        /// 64-byte serialization of `\forall i \in {0, 1}, gamma^{-1} * (beta * a_i + alpha * b_i + c_i) * H`, where `H` is the generator of `G1`.
+        gamma_abc_g1: vector<vector<u8>>,
+    }
+
+    #[resource_group_member(group = aptos_framework::zkid::Group)]
+    struct Configuration has key, store {
+        // No transaction can have more than this many zkID signatures.
+        max_zkid_signatures_per_txn: u16,
+        // How far in the future from the JWT issued at time the EPK expiry can be set.
+        max_exp_horizon: u64,
+        // The training wheels PK, if training wheels are on
+        training_wheels_pubkey: Option<vector<u8>>,
+    }
+
+    // genesis.move needs to initialize the devnet VK
+    friend aptos_framework::genesis;
+
+    public(friend) fun initialize(fx: &signer, vk: Groth16VerificationKey, constants: Configuration) {
+        system_addresses::assert_aptos_framework(fx);
+
+        move_to(fx, vk);
+        move_to(fx, constants);
+    }
+
+    #[test_only]
+    public fun initialize_for_test(fx: &signer, vk: Groth16VerificationKey, constants: Configuration) {
+        initialize(fx, vk, constants)
+    }
+
+    public fun new_groth16_verification_key(alpha_g1: vector<u8>, beta_g2: vector<u8>, gamma_g2: vector<u8>, delta_g2: vector<u8>, gamma_abc_g1: vector<vector<u8>>): Groth16VerificationKey {
+        Groth16VerificationKey {
+            alpha_g1,
+            beta_g2,
+            gamma_g2,
+            delta_g2,
+            gamma_abc_g1,
+        }
+    }
+
+    /// Returns the Groth16 VK for our devnet deployment.
+    public fun devnet_groth16_vk(): Groth16VerificationKey {
```

Comment:
net-specific large constants may unnecessarily grow framework size. (We hit framework upgrade errors in randomnet due to too large framework).

We probably want to make it data, not code.

Reply:
How? It needs to be initialized in `genesis.move` for devnet...

Reply:
i thought we will run a governance txn to
- turn on jwk
- turn on zkid
- put in vk and constants
- put in oidc providers
- reconfig

Reply:
if you have to hardcode some value for devnet, i think we them go to rust, not in move.

Reply:
e.g., [here](https://github.com/aptos-labs/aptos-core/blob/c27264d7ca0ec1023ce35934a323b062e0ce82ce/aptos-move/vm-genesis/src/lib.rs#L473) is how features resource is initialized in genesis.

Reply:
Features are initialized that way because the ground truth for them is in Rust in `default_features`.

For zkID, the ground truth need not be in Rust AFAICT. We can put it directly in `genesis.move`. 

Unless... For e2e testing in `e2e-move-tests` we might want the devnet VK to be enabled? Or maybe for the CLI?

Not sure how to think about it; we'll have to ask around. For now, this will do.